### ISSUE TITLE
Reproduce Settings

### ISSUE THREAD

**Issue by v-manhlt3** [jaeyeonkim99/EnCLAP#7]
Hi, 

First of all, this project is great, and I really appreciate that you open-sourced the pre-trained model and code. I am trying to retrain the `` BART-large`` model by using the source code with ``batch-size=64`` using two GPUs. However, the best performance I can get is:
``{'meteor': 0.2538, 'cider_d': 0.7668, 'spice': 0.1846, 'spider': 0.4757} ``

How many GPUs and the batch-size of your pretrained model do you use for training your ``BART-large`` model?

**Comment by jaeyeonkim99** [comment]
Hello! Thank you for trying out.

We used the batch size of 64 for the large model on AudioCaps. We used 4 GPUs to train, but believe 2 will be enough to train the model.

To give more details regarding the reproduction, we used `bf-16` for mixed precision, and checkpoints around epochs 12-13 gave us the best result. You may evaluate several more checkpoints, or train again with different seeds to get improved 
 results. Also during hyperparameter tuning, we observed that results fluctuated a bit based on the learning rate, so changing the learning rate slightly (e.g. 2.5e-5, 2e-5 instead of 3e-5) may improve the performance on your training setup.

Additionally, the codebase is reimplemented from the original experiment for release with some changes. We reproduced `BART-base` results almost identically, but didn't check the large ones yet. We will notify if there exists any factor that may affect reproducibility.

**Comment by v-manhlt3** [comment]
Hi,

Thanks for your prompt response. I have been trying to reproduce the result, but I am not able to reach the result of your pretrained checkpoint. So I guess that 4 GPUs is a necessary condition to reproduce SOTA performance.

Cheers

**Comment by jaeyeonkim99** [comment]
Maybe different numbers of the wav files in the dataset may affect the results, since AudioCaps is based on Youtube IDs and available files always change.

https://github.com/XinhaoMei/ACT

We used the version of AudioCaps dataset from the above repository.

**Comment by v-manhlt3** [comment]
Hi,

I am trying to reproduce results from the Clotho dataset, but we do not know which version of the Clotho dataset used in the paper. Could you please let us know which version of cloth is used in your paper?

**Comment by jaeyeonkim99** [comment]
Hello!

We used version 2.1 of Clotho.

Also, we recommend to use the code from branch `reproduction-issue`. We find out there was a difference in masking strategy from original implementation used for the experiment, and fixed it.

**Comment by v-manhlt3** [comment]
Thank you so much for your clarification!
